ingester/src/arcmap.rs

```diff
@@ -36,9 +36,11 @@ impl<K, V, S> std::ops::Deref for ArcMap<K, V, S> {
 
 impl<K, V> Default for ArcMap<K, V> {
     fn default() -> Self {
+        let map: HashMap<K, Arc<V>> = Default::default();
+        let hasher = map.hasher().clone();
         Self {
-            map: Default::default(),
-            hasher: Default::default(),
+            map: RwLock::new(map),
+            hasher,
```

Comment:
I suggest a comment as it is so subtle to try and help future readers. Something like

```suggestion
            // The same hasher should be used by everything that hashes for a consistent result.
            // otherwise some threads may not see the same contents
            // see https://github.com/influxdata/influxdb_iox/pull/6086 for details
            hasher,
```

Not sure if there is a better ticket reference